### ISSUE TITLE
How should the group update documents in 'TR' space?

### ISSUE THREAD

**Issue by halindrome** [w3c/webpayments#167]
TL;DR: There is a way to automatically update documents in the W3C formal publication space and I am curious if the group is open to doing that with our documents once they are formally published.

---

The W3C publication process is somewhat complex.  At the end of the day, formal publications are put up on the W3C site under the /TR/ tree using a "short name" that is approved by the domain lead.  (For the HTTP API it might be something like https://w3.org/TR/payment-http-api).  Initial publication has a lot of [process ](http://www.w3.org/Consortium/Process/) around it (as does any document 'transition').  After that, up until it reaches the status of "Candidate Recommendation", a working group can update the draft as often as they like. If a document is a "Note" it can also be updated at any time.

There are two schools of thought on how often to update:

1) Update periodically (e.g., every 2 months) or when there are significant changes (e.g., all issues are resolved, a new section is added).
2) Update every time the editor(s) merge a change into the main branch.

As with most things in life, there are trade-offs:

With option 1, there is usually a 'call for consensus' in the group to update the publication.  When there are many documents being managed (as in our case) this can be a little burdensome.  However, it means that there are obvious milestones where the group can direct the public to review and comment on things.  Publication in this sort of model is more of an 'event' and often engenders blog posts, tweets, etc.

With option 2, there can be a lot churn in the document that is in TR space.  Also, if there is not a good 'diff-marked' version or a good change log, it can be difficult for reviewers who are not active in the group to evaluate changes.  If it happens often, it is less of an 'event' and people in the wild seem to take less notice of the publication.

To date we have been using option 1 (or a variant of it).  I feel like we have been doing this by default, rather than taking an active decision on which option to use.

So I have opened this issue to ask for a decision of the working group: Should we be publishing every time an editor merges changes, or should we be publishing only when we want there to be an 'event'?

**Comment by halindrome** [comment]
My position on this is that our default position should be that we only publish when we want there to be an event.  If, for a specific document, we want it to be "living" and update often, we should make an explicit decision to enable that for that document.  The new "Web Payments Overview" might be one such document, for example.

**Comment by msporny** [comment]
> Should we be publishing every time an editor merges changes, or should we be publishing only when we want there to be an 'event'?

Many of the actively developed specifications in TR space are out of date. I'd like to see if we can fix that by doing continuous publishing of EDs to TR space. For 'events', we can create time-stamped versions. Here are some scenarios:
1. Editors want to publish a new ED to TR-space - they just do it as an ED.
2. WG wants to published a reviewed version and make some noise about the release - WG reviews spec and the document is published as a time-stamped WD to TR space.

Using this approach, we keep TR-space at the latest version of the specs /and/ we are also afforded the ability to refer people to time-stamped versions of these documents. I think that gets us the best of both worlds, right?

**Comment by ianbjacobs** [comment]
I believe we have already resolved to publish according to a certain process:
 https://www.w3.org/2016/04/14-wpwg-minutes#item02

We have not yet implemented it. This is linked from the how we work page:
 https://github.com/w3c/webpayments/wiki/How-the-Working-Group-works

If that page is out-of-date, let's update it.

ian

**Comment by webpayments** [comment]
Ian, Thank you for the reminder that we had discussed this back in April. I
had obviously forgotten.

Manu, I don't think there are EDs in TR space.  The TR pages have a concept
of a nightly draft update - but I have no idea what that might be.

On Mon, Jul 11, 2016 at 10:52 PM, ianbjacobs notifications@github.com
wrote:

> I believe we have already resolved to publish according to a certain
> process:
> https://www.w3.org/2016/04/14-wpwg-minutes#item02
> 
> We have not yet implemented it. This is linked from the how we work page:
> https://github.com/w3c/webpayments/wiki/How-the-Working-Group-works
> 
> If that page is out-of-date, let's update it.
> 
> ian
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/w3c/webpayments/issues/167#issuecomment-231930751,
> or mute the thread
> https://github.com/notifications/unsubscribe/APFrHOkw0SX9YeeB537773AiDTSOfe19ks5qUw-OgaJpZM4JJlGh
> .

## 

Shane McCarron
Projects Manager, Spec-Ops

**Comment by adrianhopebailie** [comment]
My recollection of the agreed process is that editors will push to TR space through auto-publishing from a named branch or similar

**Comment by webpayments** [comment]
Yes, and that is reflected in the wiki.  Apologies for not remembering we
had reached this agreement.  I note, however, that we have not yet actually
implemented this.  Let's pick a spec to be a guinea pig!  I nominate the
'Web Payments Overview' document, which I expect will be a note.  That one
is going to churn as the overview matures anyway so it is an ideal
candidate.

On Tue, Jul 12, 2016 at 6:36 AM, Adrian Hope-Bailie <
notifications@github.com> wrote:

> My recollection of the agreed process is that editors will push to TR
> space through auto-publishing from a named branch or similar
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/w3c/webpayments/issues/167#issuecomment-232017524,
> or mute the thread
> https://github.com/notifications/unsubscribe/APFrHPCKuCdy2LCAlBOsxSO1F0DWuQO1ks5qU3wigaJpZM4JJlGh
> .

## 

Shane McCarron
Projects Manager, Spec-Ops

**Comment by adrianhopebailie** [comment]
> I nominate the 'Web Payments Overview' document

+1

**Comment by ianbjacobs** [comment]
Closed as of:
 https://lists.w3.org/Archives/Public/public-payments-wg/2016Dec/0011.html
